CMaxTable.lua

```diff
@@ -19,8 +19,7 @@ end
 
 function CMaxTable:updateGradInput(input, gradOutput)
    for i=1,#input do
-      self.gradInput[i] = torch.Tensor()
-      self.gradInput[i]:resizeAs(input[i]):fill(0.0)
+      self.gradInput[i] = input[i]:clone():fill(0.0)
```

Comment:
I agree the repeated memory allocation seems pretty bad. I'm not sure `.new()` works here though - I thought this just creates an empty Tensor, and the sizes need to match. Please do correct me if I'm wrong though, or if there's any other ways of implementing this.